app/products/calls/components/captions.tsx

```diff
@@ -0,0 +1,119 @@
+        position: 'relative',
+        width: '90%',
+        height: 48,
+    },
+    captionContainer: {
+        display: 'flex',
+        height: 400,
+        bottom: -352, // 48-400, to place the bottoms at the same place
+        gap: 8,
+        alignItems: 'center',
+        flexDirection: 'column-reverse',
+        overflow: 'hidden',
+    },
+    caption: {
+        paddingTop: 1,
+        paddingRight: 8,
+        paddingBottom: 3,
+        paddingLeft: 8,
+        borderRadius: 4,
+        backgroundColor: changeOpacity('#000', 0.64),
+    },
+    captionNotice: {
+        display: 'flex',
+        flexDirection: 'row',
+        gap: 8,
+    },
+    text: {
+        color: '#FFF',
+        fontFamily: 'Open Sans',
+        fontSize: 16,
+        fontStyle: 'normal',
+        fontWeight: '400',
+        lineHeight: 22,
+        textAlign: 'center',
+    },
+});
+
+type Props = {
+    captionsDict: Dictionary<LiveCaptionMobile>;
+    sessionsDict: Dictionary<CallSession>;
+    teammateNameDisplay: string;
+}
+
+const Captions = ({captionsDict, sessionsDict, teammateNameDisplay}: Props) => {
+    const intl = useIntl();
+    const [showCCNotice, setShowCCNotice] = useState(true);
+
+    useEffect(() => {
+        const timeoutID = setTimeout(() => {
+            setShowCCNotice(false);
+        }, 2000);
+        return () => clearTimeout(timeoutID);
+    }, []);
+
+    const captionsArr = Object.values(captionsDict).reverse();
+
+    if (showCCNotice && captionsArr.length > 0) {
+        setShowCCNotice(false);
+    }
+    if (showCCNotice) {
+        return (
+            <View style={styles.spacingContainer}>
+                <View style={styles.captionContainer}>
+                    <View style={[styles.caption, styles.captionNotice]}>
+                        <CompassIcon
+                            name='closed-caption-outline'
+                            color={'#FFF'}
+                            size={18}
+                            style={{alignSelf: 'center'}}
+                        />
+                        <FormattedText
+                            id={'mobile.calls_captions_turned_on'}
+                            defaultMessage={'Live captions turned on'}
+                            style={styles.text}
+                        />
+                    </View>
+                </View>
+            </View>
+        );
+    }
+
+    return (
+        <View style={styles.spacingContainer}>
+            <View style={styles.captionContainer}>
+                {captionsArr.map((cap) => (
+                    <View
+                        key={cap.captionId}
+                        style={styles.caption}
+                    >
+                        <Text
+                            style={styles.text}
+                            numberOfLines={0}
+                        >
+                            {`(${displayUsername(sessionsDict[cap.sessionId]?.userModel, intl.locale, teammateNameDisplay)}) ${cap.text}`}
```

Comment:
Again more of a question for my own understanding. Is there any downside in using the whole sessions dictionary in terms of re-renders? After all what we really need is just the participant's id, not even the session as technically the user id would be sufficient right?

Reply:
Not sure what you mean here about 'all what we really need is just the participant's id' -- it's the userModel that we need. If that's what you meant, then I guess we could create a userModels dict in the parent component and only pass that in, but that wouldn't save any renders because it would be a new object also on every parent render, and we would then be creating another new object that would need to be collected later.

Reply:
My point here is that the sessions and participants state is data that can mutate quite often whereas the users (participants) models would only change when joining/leaving. So I was wondering whether there could be value in making this component depend on the latter.

Reply:
Normally yes, that would be the right thing to do. But the way that the userModels are stored, we would need to extract it out on any change anyway. That would potentially be a good optimization to do (make an independent useModel dict), so I'll put that on my list.  On the other hand, the amount of speedup in this case would be very little I imagine, given how it's only a few lines of captions. But the principle is good for maybe the participants grid or something more heavy.